src/Microsoft.AspNetCore.WebHooks.Receivers/Metadata/WebHookBodyType.cs

```diff
@@ -1,30 +1,52 @@
+    /// <remarks>The value <c>0</c> is not valid; a bit must be set in any use of this type.</remarks>
+    /// <seealso cref="IWebHookBodyTypeMetadata.BodyType"/>
+    /// <seealso cref="IWebHookBodyTypeMetadataService.BodyType"/>
+    [Flags]
     public enum WebHookBodyType
     {
         /// <summary>
-        /// Request must have <c>content-type</c> <c>application/x-www-form-urlencoded</c>. A bound <c>data</c>
-        /// parameter should have an associated <c>[FromForm]</c> attribute.
+        /// Request must have <c>content-type</c> <c>application/x-www-form-urlencoded</c>. The
+        /// <see cref="ApplicationModels.WebHookModelBindingProvider"/> gives a bound <c>data</c> parameter the same
+        /// <see cref="Mvc.ModelBinding.BindingInfo"/> as an associated <c>[FromForm]</c> attribute when the action or
+        /// receiver has this <see cref="WebHookBodyType"/>.
         /// </summary>
         Form = 1,
 
         /// <summary>
-        /// Request must have <c>content-type</c> <c>application/json</c>, <c>text/json</c>, or a subset. A bound
-        /// <c>data</c> parameter should have an associated <c>[FromBody]</c> attribute.
+        /// Request must have <c>content-type</c> <c>application/json</c>, <c>application/*+json</c>, <c>text/json</c>,
+        /// or a subset. The <see cref="ApplicationModels.WebHookModelBindingProvider"/> gives a bound <c>data</c>
+        /// parameter the same <see cref="Mvc.ModelBinding.BindingInfo"/> as an associated <c>[FromBody]</c> attribute
+        /// when the action or receiver has this <see cref="WebHookBodyType"/>.
+        /// </summary>
+        Json = 2,
+
+        /// <summary>
+        /// Request must have <c>content-type</c> <c>application/xml</c>, <c>application/*+xml</c>, <c>text/xml</c>,
+        /// or a subset. The <see cref="ApplicationModels.WebHookModelBindingProvider"/> gives a bound <c>data</c>
+        /// parameter the same <see cref="Mvc.ModelBinding.BindingInfo"/> as an associated <c>[FromBody]</c> attribute
+        /// when the action or receiver has this <see cref="WebHookBodyType"/>.
         /// </summary>
-        Json,
+        Xml = 4,
 
         /// <summary>
-        /// Request must have <c>content-type</c> <c>application/xml</c>, <c>text/xml</c>, or a subset. A bound
-        /// <c>data</c> parameter should have an associated <c>[FromBody]</c> attribute.
+        /// Request may have any supported <c>content-type</c>. The
+        /// <see cref="ApplicationModels.WebHookModelBindingProvider"/> ignores bound <c>data</c> parameters when the
+        /// action or receiver has this <see cref="WebHookBodyType"/>.
         /// </summary>
-        Xml,
+        /// <remarks>
+        /// This value is intended for use as <see cref="GeneralWebHookAttribute"/>'s default
+        /// <see cref="IWebHookBodyTypeMetadata.BodyType"/>.
+        /// </remarks>
+        All = Form | Json | Xml,
```

Comment:
BTW it would be a breaking change to add a new value to this enum because the value of `All` would likely change, and these are all `const`s.

Reply:
Yup. Fortunately, this is the entire set of body types known in https://github.com/aspnet/AspNetWebHooks and I'm not aware of other senders that do anything different.